lib/improver/utilities/temporal.py

```diff
@@ -365,3 +365,154 @@ def get_forecast_times(forecast_length, forecast_date=None,
                        range(51, forecast_length+1, 3)])
 
     return forecast_times
+
+
+class TemporalInterpolation(object):
+
+    """
+    Interpolate data to intermediate times between the validity times of two
+    cubes. This can be used to fill in missing data (e.g. for radar fields) or
+    to ensure data is available at the required intervals when model data is
+    not available at these times.
+    """
+
+    def __init__(self, interval_in_minutes=None, times=None):
+        """
+        Initialise class.
+
+        Keyword Args:
+            interval_in_minutes (int):
+                Specifies the interval in minutes at which to interpolate
+                between the two input cubes. A number of minutes which does not
+                divide up the interval equally will raise an exception::
+
+                    e.g. cube_t0 valid at 03Z, cube_t1 valid at 06Z,
+                    interval_in_minutes = 60 --> interpolate to 04Z and 05Z.
+
+            times (list or tuple of datetime.datetime objects):
+                A list of datetime objects specifying the times to which to
+                interpolate.
+        """
+        if interval_in_minutes is None and times is None:
+            raise ValueError("TemporalInterpolation: One of "
+                             "'interval_in_minutes' or 'times' must be set. "
+                             "Currently both are none.")
+
+        self.interval_in_minutes = interval_in_minutes
+        self.times = times
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<TemporalInterpolation: interval_in_minutes: {}, '
+                  'times: {}>')
+        return result.format(self.interval_in_minutes, self.times)
+
+    def construct_time_list(self, initial_time, final_time):
+        """
+        A function to construct a list of datetime objects formatted
+        appropriately for use by iris' interpolation method.
+
+        Args:
+            initial_time (datetime.datetime):
+                The start of the period over which a time list is to be
+                constructed.
+            final_time (datetime.datetime).
+                The end of the period over which a time list is to be
+                constructed.
+
+        Returns:
+            time_list (list of tuples):
+                A list of tuples formatted suitably for use by iris'
+                interpolation method::
+
+                    e.g. [('time', <datetime object 0>),
+                          ('time', <datetime object 1>), ...]
+        Raises:
+            ValueError: If list of times provided falls outside the range
+                        specified by the initial and final times.
+            ValueError: If the interval_in_minutes does not divide the time
+                        range up equally.
+        """
+        time_list = []
+        if self.times is not None:
+            self.times.sort()
```

Comment:
You're `__init__` docstring says `times` can be a `list` or a `tuple` but I don't think tuples have a `.sort` method.

Reply:
Good spot, changed to use ```self.times = sorted(self.times)```